server/methods/core/cart.js

```diff
@@ -671,6 +671,11 @@ Meteor.methods({
     order.shipping[0].items.shipped = false;
     order.shipping[0].items.delivered = false;
 
+    // begin a new shipping workflow for the order
+    order.shipping[0].workflow = {
+      status: "new",
+      workflow: ["coreOrderWorkflow/notStarted"]
+    };
     order.billing[0].currency.exchangeRate = exchangeRate;
```

Comment:
This issue seems like it is older than this particular PR because this code is older, but needs to get solved ASAP.

Can't assume that this is the only payment in this order, or that the 0th payment corresponds with this shipment.

I'll repeat myself a few more times: hardcoded 0th indexes are an indicator that code is not ready for marketplace.

Hard coded indexes, 0th especially, in general are a code smell. It's a sign that that the data structure doesn't fit the application, or that you don't have a better way to identify elements in the array other than just arbitrarily taking the first one.